src/EVM/Expr.hs

```diff
@@ -1057,3 +1058,52 @@ evalProp prop =
       | l == r = PBool True
       | otherwise = o
     go p = p
+
+
+data ConstState = ConstState
+  { values :: Map.Map (Expr EWord) W256
+  , canBeSat :: Bool
+  }
+  deriving (Show)
+
+-- | Folds constants
+constFoldProp :: [Prop] -> ConstState
```

Comment:
I think we could just return `canBeSat` here?

Reply:
Oh, yes, of course! Good idea :) Fixed in 88381efaaa93df187691d2cb404d8c8598ec5722